### PR TITLE
265 execute command in terminal not working in uefi shell

### PR DESCRIPTION
**Pull request by JanPrusinowski**
[https://jira-3mdeb.atlassian.net/browse/PROT-1302](url)

### CODE REVIEW

File: lib/terminal.robot [hunk 1]
```diff
@@ -160,3 +160,13 @@ Execute Command In Terminal
     # Drop last newline, if any
     ${output}=    Strip String    ${output}    mode=right    characters=\n\r
     RETURN    ${output}
+
+Execute Shell Command
+    [Documentation]    Universal keyword to execute command in Shell.
```
Comment (macpijan) [inline lib/terminal.robot:165]: Is it really universal, or meant for UEFI shell? If meant for UEFI shell, I would rename the keyword to reflect that

**Review by macpijan — CHANGES_REQUESTED** [review]
Testing:

1. New self-test works:

```
robot  -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no self-tests/terminal.robot
==============================================================================
Terminal :: This suite verifies the correct operation of keywords entering ...
==============================================================================
Execute UEFI Shell Command :: Test Execute Shell Command kwd          | PASS |
------------------------------------------------------------------------------
Terminal :: This suite verifies the correct operation of keywords ... | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Output:  /home/macpijan/projects/github/dasharo/open-source-firmware-validation/output.xml
Log:     /home/macpijan/projects/github/dasharo/open-source-firmware-validation/log.html
Report:  /home/macpijan/projects/github/dasharo/open-source-firmware-validation/report.html
```

2. Modified Secure Boot suite does not work:

```
robot  -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no dasharo-security/secure-boot.robot
==============================================================================
Secure-Boot                                                                   
==============================================================================
SBO001.001 Check Secure Boot default state (firmware) :: This test... | PASS |
------------------------------------------------------------------------------
SBO002.001 UEFI Secure Boot (Ubuntu 22.04) :: This test verifies t... | PASS |
------------------------------------------------------------------------------
SBO002.002 UEFI Secure Boot (Windows 11) :: This test verifies tha... | SKIP |
SBO002.002 not supported
------------------------------------------------------------------------------
SBO003.001 Attempt to boot file with the correct key from Shell (f... | FAIL |
'
FS0:\>' does not contain 'Hello, world!'
------------------------------------------------------------------------------
SBO004.001 Attempt to boot file without the key from Shell (firmwa... | FAIL |
'
FS0:\>' does not contain 'Access Denied'
------------------------------------------------------------------------------
SBO005.001 Attempt to boot file with the wrong-signed key from She... | FAIL |
'
FS0:\>' does not contain 'Access Denied'

```

it is very close, but this keyword or changes in SB lib needs some further refinement.

File: self-tests/terminal.robot [hunk 2]
```diff
@@ -0,0 +1,44 @@
+*** Settings ***
+Documentation       This suite verifies the correct operation of keywords
+...                 entering and parsing UEFI shell commands
```
[the remark this answers is not in the document]
Comment (JanPrusinowski) [reply self-tests/terminal.robot:3]: Poprawione: 32bc9ba9197fe060728781d746d4956a5ced04bd

File: lib/terminal.robot (hunk 1 above)
Comment (JanPrusinowski) [reply lib/terminal.robot:165]: changed name in: 0a78cd0aa92831ffd253a3f5d86ff71def8bcde8

**Comment by macpijan** [comment]
@JanPrusinowski I would appreciate passing logs if you say it's been fixed.
 Still fails for me:
 
 ```
 robot  -L TRACE -v config:qemu -v rte_ip:127.0.0.1 -v snipeit:no dasharo-security/secure-boot.robot
==============================================================================
Secure-Boot                                                                   
==============================================================================
SBO001.001 Check Secure Boot default state (firmware) :: This test... | PASS |
------------------------------------------------------------------------------
SBO002.001 UEFI Secure Boot (Ubuntu 22.04) :: This test verifies t... | PASS |
------------------------------------------------------------------------------
SBO002.002 UEFI Secure Boot (Windows 11) :: This test verifies tha... | SKIP |
SBO002.002 not supported
------------------------------------------------------------------------------
SBO003.001 Attempt to boot file with the correct key from Shell (f... | FAIL |
'
FS0:\>' does not contain 'Hello, world!'
------------------------------------------------------------------------------
SBO004.001 Attempt to boot file without the key from Shell (firmwa... | FAIL |
'
FS0:\>' does not contain 'Access Denied'
------------------------------------------------------------------------------
SBO005.001 Attempt to boot file with the wrong-signed key from She... | FAIL |
'
FS0:\>' does not contain 'Access Denied'
------------------------------------------------------------------------------
SBO006.001 Reset Secure Boot Keys option availability (firmware) :... | PASS |
------------------------------------------------------------------------------
SBO007.001 Attempt to boot the file after restoring keys to defaul... | FAIL |
'
FS0:\>' does not contain 'Hello, world!'
------------------------------------------------------------------------------
SBO008.001 Attempt to enroll the key in the incorrect format (firm... | PASS |
------------------------------------------------------------------------------
Secure-Boot                                                           | FAIL |
9 tests, 4 passed, 4 failed, 1 skipped
==============================================================================
Output:  /home/macpijan/projects/github/dasharo/open-source-firmware-validation/output.xml
Log:     /home/macpijan/projects/github/dasharo/open-source-firmware-validation/log.html
Report:  /home/macpijan/projects/github/dasharo/open-source-firmware-validation/report.html
```

File: self-tests/terminal.robot [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+Execute UEFI Shell Command
+    [Documentation]    Test Execute Shell Command kwd
+    Power On
+    Enter UEFI Shell
+    Execute UEFI Shell Command    map
```
Comment (macpijan) [inline self-tests/terminal.robot:36]: When reviewing another MR (https://github.com/Dasharo/open-source-firmware-validation/pull/310#discussion_r1643215693) , it struck me that this is inconsistent with other `Execute ...` keywords we have.
When we say `Execute ...` we expect the same keyword returns output as well. Can it be done? If not, we should rename it to `Write into UEFI Shell` or so, althoguh I would prefer that we can fix it to work as expected (see the `Exeucte Command In terminal` as a reference, or this RF documentation:  https://robotframework.org/robotframework/latest/libraries/Telnet.html#Execute%20Command

**Comment by macpijan** [comment]
I see some new commits @JanPrusinowski , should it work better? In such case, please provide some logs

**Pull request merged**